### ISSUE TITLE
Don't override this.props.query if already set

### ISSUE THREAD

**Issue by antonholmquist** [rackt/react-router#1648]
When a route is loaded, the get params is set to `this.props.query`.

This works well for client side code, but for server side the router has no way of getting the params, and `this.props.query` ends empty.

I would much prefer if I could choose to pass the query params to the view myself on the server side, and that the router would not override `this.props.query` (which effectively clears it). This would allow for a consistent query object both on client and server side if passed correctly on the server.

Basically, all that is needed would be by the router to leave `this.props.query` as is, if already passed from outside.

**Comment by ryanflorence** [comment]
I don't get it ... server side it gets params and query as well. Which version are you using?

**Comment by knowbody** [comment]
as @ryanflorence said both client and server get params and query
